crates/ruff/src/checkers/ast/traits.rs

```diff
@@ -0,0 +1,42 @@
+use ruff_diagnostics::Diagnostic;
```

Comment:
Nit: I prefer to name my files after the concept they solve rather than what the things are that I put in there. Doing so helps with encapsulation (use the lowest visibility), eases moving things around, and tends to be easier discoverable (I don't know that something is a trait by just reading the type name: Is `RuleContext` a trait? I don't know)